specification/security/suggest_user_profiles/types.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { UserProfileId } from '@security/_types/UserProfile'
+import { SingleKeyDictionary } from '@spec_utils/Dictionary'
+
+export class Hint {
+  /**
+   * A list of Profile UIDs to match against.
+   */
+  uids?: UserProfileId[]
+  /**
+   * A single key-value pair to match against the labels section
+   * of a profile. A profile is considered matching if it matches
+   * at least one of the strings.
+   */
+  labels?: SingleKeyDictionary<string, HintLabelValue>
```

Comment:
A single key dictionary sounds weird for a `labels` (plural) property, and I wouldn't be surprised that it's extended in the future to support multiple entries. So what about making a regular `Dictionary`?

Reply:
The use of `labels` is to mirror the User Profile data property `labels`.

I was thinking the same but checking the server code it does check that there's on element. I'll set this to `Dictionary` because like you said, I think there's a good chance we see this expand to multiple filter values.